### ISSUE TITLE
Generate a tree that contains a list of variables

### ISSUE THREAD

**Issue by AndrewC19** [DEAP/deap#665]
Hi,

I am trying to grow a syntax tree that contains a list of variables such as `[X1, X2, ..., Xn]`. My main requirement is that all of the listed variables are used somewhere in the tree (other constants can be present too).

Is it possible to use the `condition` parameter of generate to ensure that at least `n` unique variables are present in the tree?

Thanks.

**Comment by fmder** [comment]
No, currently the condition function in generate gets only the height and depth of the tree. Moreover, after generation you’d still need special crossover and mutation that keep your trees with all variables.
